src/core/api/chart_repository.go

```diff
@@ -267,16 +282,34 @@ func (cra *ChartRepositoryAPI) DeleteChartVersion() {
 	chartName := cra.GetStringFromPath(nameParam)
 	version := cra.GetStringFromPath(versionParam)
 
+	key := fmt.Sprintf("chart:%s:%s:%s", cra.namespace, chartName, version)
```

Comment:
Can we move all the code related with quota into a separate middleware or handler just like other quota handlers?

Reply:
It is difficulty to move this to middleware for calculating the resources required when upload chart package need to parse the chart info from archive file and chart info need to parse after `rewriteFileContent`. If move this to middleware, `rewriteFileContent` also need to move.